frame/gilt/src/lib.rs

```diff
@@ -96,7 +96,13 @@ pub mod pallet {
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
 		/// Currency type that this works on.
-		type Currency: ReservableCurrency<Self::AccountId>;
+		type Currency: ReservableCurrency<Self::AccountId, Balance=Self::CurrencyBalance>;
+
+		/// Just the `Currency::Balance` type; we have this item to allow us to constrain it to
+		/// `From<u64>`.
+		type CurrencyBalance:
+			sp_runtime::traits::AtLeast32BitUnsigned + codec::FullCodec + Copy
+				+ MaybeSerializeDeserialize + sp_std::fmt::Debug + Default + From<u64>;
```

Comment:
The only alternative that I can think of is to add a `where <T::Current as _>::Balance: From<u64>` to most functions of this pallet, and while this is not pretty either, I guess it has less footprint than sprinkling `where`s.